src/sentry/replays/consumers/recording.py

```diff
@@ -39,17 +41,50 @@ class ProcessReplayRecordingStrategyFactory(ProcessingStrategyFactory[KafkaPaylo
     chunks.
     """
 
+    def __init__(
+        self,
+        input_block_size: int,
+        max_batch_size: int,
+        max_batch_time: int,
+        num_processes: int,
+        output_block_size: int,
+    ) -> None:
+        self.input_block_size = input_block_size
+        self.max_batch_size = max_batch_size
+        self.max_batch_time = max_batch_time
+        self.num_processes = num_processes
+        self.output_block_size = output_block_size
+        self.use_multi_proc = self.num_processes > 1
+
     def create_with_partitions(
         self,
         commit: Commit,
         partitions: Mapping[Partition, int],
     ) -> Any:
-        step = RunTaskInThreads(
-            processing_function=move_replay_to_permanent_storage,
-            concurrency=4,
-            max_pending_futures=50,
-            next_step=CommitOffsets(commit),
-        )
+        # Give mypy some help.
+        step: Union[
+            RunTaskWithMultiprocessing[MessageContext, Any], RunTaskInThreads[MessageContext, Any]
+        ]
+
+        if self.use_multi_proc:
+            step = RunTaskWithMultiprocessing(
+                function=move_replay_to_permanent_storage,
+                next_step=CommitOffsets(commit),
+                num_processes=self.num_processes,
```

Comment:
would be helpful for future if there were some kind of comment or link to docs / arroyo source that explains these options

Reply:
Agreed though the docs aren't totally clear either.

Reply:
This doc page is super helpful: https://getsentry.github.io/arroyo/strategies/run_task_with_multiprocessing.html

Reply:
i think those options are sufficiently documented nowadays (happened after colleen's shipped email)

we do have some tickets open in arroyo to have better defaults for them